contracts/DepositContract.sol

```diff
@@ -0,0 +1,75 @@
+pragma solidity ^0.4.22;
+
+contract DepositContract {
+
+  mapping(address => uint) private Stakes;
+  mapping(address => uint) private Indices;
+  address[] private nodeAddressList;
+
+  event LogNewNode(address indexed nodeAddress, uint index, uint stake);
+  event LogUpdateStake(address indexed nodeAddress, uint index,  uint stake);
+  event LogDeleteNode(address indexed nodeAddress, uint index);
+  
+  function isNode(address nodeAddress) public constant returns(bool isIndeed) 
+  {
+    if(nodeAddressList.length == 0) return false;
+    return (nodeAddressList[Indices[nodeAddress]] == nodeAddress);
+  }
+
+function deposit() public payable returns(uint index)
+  {
+    if(isNode(msg.sender)) {
+    	Stakes[msg.sender] += msg.value;
+    	emit LogUpdateStake(msg.sender, Indices[msg.sender], Stakes[msg.sender]);
```

Comment:
why emit these events?

Reply:
The compiler was showing warnings without `emit`.

Reply:
That's weird. My faucet contract never have this problem. You can try remix. I used that before and it worked fine.

Reply:
Yes I used remix. I used 4.24 compiler version. 
This is what happened.
`browser/DepositContract.sol:56:5: Warning: Invoking events without "emit" prefix is deprecated.
    LogDeleteNode(nodeAddress,indexToDelete);`